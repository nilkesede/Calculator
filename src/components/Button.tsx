import React from 'react'
import styled from 'styled-components/native'

interface WrapperProps {
  readonly big?: boolean
  readonly dark?: boolean
  readonly primary?: boolean
}

interface ButtonProps extends WrapperProps {
  readonly value: string
  readonly onPress: (command: string) => void
}

const Wrapper = styled.TouchableOpacity<WrapperProps>`
  background-color: ${({theme}) => theme.colors.defaultGray}
  justify-content: center;
  width: 25%;
  border: 1px solid ${props => props.theme.colors.backgroundColor};

  ${props => props.dark && `background-color: ${props.theme.colors.darkGray}`}
  ${props => props.primary && `background-color: ${props.theme.colors.primaryColor}`}
  ${({big}) => big && 'width: 50%;'}
`

const Text = styled.Text`
  font-weight: 200;
  font-size: 45px;
  color: white;
  text-align: center;
`

export default function Button({value, onPress, big = false, dark = false, primary = false}: ButtonProps) {
  return (
    <Wrapper big={big} dark={dark} primary={primary} onPress={() => onPress(value)}>
      <Text adjustsFontSizeToFit numberOfLines={1}>
        {value}
      </Text>
    </Wrapper>
  )
}
