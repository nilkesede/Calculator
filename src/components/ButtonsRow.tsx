import React from 'react'
import styled from 'styled-components/native'

const Wrapper = styled.View`
  flex: 1;
  flex-direction: row;
`
export default function ButtonsRow({children}: {children: JSX.Element[]}) {
  return <Wrapper>{children}</Wrapper>
}
