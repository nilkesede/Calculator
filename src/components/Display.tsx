import React from 'react'
import styled from 'styled-components/native'

const Wrapper = styled.View`
  background-color: #303030;
  flex: 1;
  justify-content: flex-end;
`

const Text = styled.Text`
  font-size: 100px;
  text-align: right;
  font-weight: 100;
  color: white;
`

export default function Display({value}: {value: string}) {
  return (
    <Wrapper>
      <Text adjustsFontSizeToFit numberOfLines={1}>
        {value}
      </Text>
    </Wrapper>
  )
}
