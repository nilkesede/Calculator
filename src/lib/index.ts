export {default as Processor} from './processor'
export {default as useProcessor} from './use-processor'
