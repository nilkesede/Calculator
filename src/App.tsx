import React, {useState, useCallback} from 'react'
import {StatusBar} from 'react-native'
import styled, {ThemeProvider} from 'styled-components/native'

import {theme} from './config'
import {Display, Keyboard} from './components'

const View = styled.View`
  flex: 1;
`

export default function App() {
  const [value, setValue] = useState('')

  const processor = useCallback((command: string) => {
    setValue(command)
  }, [])

  return (
    <ThemeProvider theme={theme}>
      <StatusBar barStyle="light-content" />
      <View>
        <Display value={value} />
        <Keyboard callback={processor} />
      </View>
    </ThemeProvider>
  )
}
