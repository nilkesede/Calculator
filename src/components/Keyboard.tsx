import React from 'react'
import styled from 'styled-components/native'

import ButtonsRow from './ButtonsRow'
import Button from './Button'

const Wrapper = styled.View`
  background-color: ${props => props.theme.colors.backgroundColor};
  flex: 2;
`
export default function Keyboard({callback}: {callback: (command: string) => void}) {
  return (
    <Wrapper>
      <ButtonsRow>
        <Button dark value={'AC'} onPress={callback} />
        <Button dark value={'+/-'} onPress={callback} />
        <Button dark value={'%'} onPress={callback} />
        <Button primary value={'/'} onPress={callback} />
      </ButtonsRow>
      <ButtonsRow>
        <Button value={'7'} onPress={callback} />
        <Button value={'8'} onPress={callback} />
        <Button value={'9'} onPress={callback} />
        <Button primary value={'x'} onPress={callback} />
      </ButtonsRow>
      <ButtonsRow>
        <Button value={'4'} onPress={callback} />
        <Button value={'5'} onPress={callback} />
        <Button value={'6'} onPress={callback} />
        <Button primary value={'-'} onPress={callback} />
      </ButtonsRow>
      <ButtonsRow>
        <Button value={'1'} onPress={callback} />
        <Button value={'2'} onPress={callback} />
        <Button value={'3'} onPress={callback} />
        <Button primary value={'+'} onPress={callback} />
      </ButtonsRow>
      <ButtonsRow>
        <Button big value={'0'} onPress={callback} />
        <Button value={'.'} onPress={callback} />
        <Button primary value={'='} onPress={callback} />
      </ButtonsRow>
    </Wrapper>
  )
}
