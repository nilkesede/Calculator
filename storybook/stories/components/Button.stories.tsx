import React from 'react'
import {storiesOf} from '@storybook/react-native'

import {Button} from '../../../src/components'

storiesOf('Button', module)
  .add('default', () => <Button value={'OK'} />)
  .add('big', () => <Button big value={'OK'} />)
  .add('dark', () => <Button dark value={'OK'} />)
  .add('primary', () => <Button primary value={'OK'} />)
