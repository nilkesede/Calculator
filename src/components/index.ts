import Button from './Button'
import ButtonsRow from './ButtonsRow'
import Display from './Display'
import Keyboard from './Keyboard'

export {Button, ButtonsRow, Display, Keyboard}
