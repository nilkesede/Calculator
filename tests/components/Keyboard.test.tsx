import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

import {Keyboard} from '../../src/components'

it('renders correctly', () => {
  const tree = renderer.create(<Keyboard />).toJSON()
  expect(tree).toMatchSnapshot()
})
