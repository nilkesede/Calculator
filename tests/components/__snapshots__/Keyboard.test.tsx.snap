// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#303030",
        "flexBasis": 0,
        "flexGrow": 2,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#525252",
          "borderColor": "#303030",
          "borderStyle": "solid",
          "borderWidth": 1,
          "justifyContent": "center",
          "opacity": 1,
          "width": "25%",
        }
      }
    >
      <Text
        adjustsFontSizeToFit={true}
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 45,
              "fontWeight": "200",
              "textAlign": "center",
            },
          ]
        }
      >
        AC
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#525252",
          "borderColor": "#303030",
          "borderStyle": "solid",
          "borderWidth": 1,
          "justifyContent": "center",
          "opacity": 1,
          "width": "25%",
        }
      }
    >
      <Text
        adjustsFontSizeToFit={true}
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 45,
              "fontWeight": "200",
              "textAlign": "center",
            },
          ]
        }
      >
        +/-
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#525252",
          "borderColor": "#303030",
          "borderStyle": "solid",
          "borderWidth": 1,
          "justifyContent": "center",
          "opacity": 1,
          "width": "25%",
        }
      }
    >
      <Text
        adjustsFontSizeToFit={true}
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 45,
              "fontWeight": "200",
              "textAlign": "center",
            },
          ]
        }
      >
        %
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#fa9f0d",
          "borderColor": "#303030",
          "borderStyle": "solid",
          "borderWidth": 1,
          "justifyContent": "center",
          "opacity": 1,
          "width": "25%",
        }
      }
    >
      <Text
        adjustsFontSizeToFit={true}
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 45,
              "fontWeight": "200",
              "textAlign": "center",
            },
          ]
        }
      >
        /
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#707070",
          "borderColor": "#303030",
          "borderStyle": "solid",
          "borderWidth": 1,
          "justifyContent": "center",
          "opacity": 1,
          "width": "25%",
        }
      }
    >
      <Text
        adjustsFontSizeToFit={true}
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 45,
              "fontWeight": "200",
              "textAlign": "center",
            },
          ]
        }
      >
        7
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#707070",
          "borderColor": "#303030",
          "borderStyle": "solid",
          "borderWidth": 1,
          "justifyContent": "center",
          "opacity": 1,
          "width": "25%",
        }
      }
    >
      <Text
        adjustsFontSizeToFit={true}
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 45,
              "fontWeight": "200",
              "textAlign": "center",
            },
          ]
        }
      >
        8
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#707070",
          "borderColor": "#303030",
          "borderStyle": "solid",
          "borderWidth": 1,
          "justifyContent": "center",
          "opacity": 1,
          "width": "25%",
        }
      }
    >
      <Text
        adjustsFontSizeToFit={true}
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 45,
              "fontWeight": "200",
              "textAlign": "center",
            },
          ]
        }
      >
        9
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#fa9f0d",
          "borderColor": "#303030",
          "borderStyle": "solid",
          "borderWidth": 1,
          "justifyContent": "center",
          "opacity": 1,
          "width": "25%",
        }
      }
    >
      <Text
        adjustsFontSizeToFit={true}
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 45,
              "fontWeight": "200",
              "textAlign": "center",
            },
          ]
        }
      >
        x
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#707070",
          "borderColor": "#303030",
          "borderStyle": "solid",
          "borderWidth": 1,
          "justifyContent": "center",
          "opacity": 1,
          "width": "25%",
        }
      }
    >
      <Text
        adjustsFontSizeToFit={true}
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 45,
              "fontWeight": "200",
              "textAlign": "center",
            },
          ]
        }
      >
        4
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#707070",
          "borderColor": "#303030",
          "borderStyle": "solid",
          "borderWidth": 1,
          "justifyContent": "center",
          "opacity": 1,
          "width": "25%",
        }
      }
    >
      <Text
        adjustsFontSizeToFit={true}
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 45,
              "fontWeight": "200",
              "textAlign": "center",
            },
          ]
        }
      >
        5
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#707070",
          "borderColor": "#303030",
          "borderStyle": "solid",
          "borderWidth": 1,
          "justifyContent": "center",
          "opacity": 1,
          "width": "25%",
        }
      }
    >
      <Text
        adjustsFontSizeToFit={true}
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 45,
              "fontWeight": "200",
              "textAlign": "center",
            },
          ]
        }
      >
        6
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#fa9f0d",
          "borderColor": "#303030",
          "borderStyle": "solid",
          "borderWidth": 1,
          "justifyContent": "center",
          "opacity": 1,
          "width": "25%",
        }
      }
    >
      <Text
        adjustsFontSizeToFit={true}
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 45,
              "fontWeight": "200",
              "textAlign": "center",
            },
          ]
        }
      >
        -
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#707070",
          "borderColor": "#303030",
          "borderStyle": "solid",
          "borderWidth": 1,
          "justifyContent": "center",
          "opacity": 1,
          "width": "25%",
        }
      }
    >
      <Text
        adjustsFontSizeToFit={true}
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 45,
              "fontWeight": "200",
              "textAlign": "center",
            },
          ]
        }
      >
        1
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#707070",
          "borderColor": "#303030",
          "borderStyle": "solid",
          "borderWidth": 1,
          "justifyContent": "center",
          "opacity": 1,
          "width": "25%",
        }
      }
    >
      <Text
        adjustsFontSizeToFit={true}
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 45,
              "fontWeight": "200",
              "textAlign": "center",
            },
          ]
        }
      >
        2
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#707070",
          "borderColor": "#303030",
          "borderStyle": "solid",
          "borderWidth": 1,
          "justifyContent": "center",
          "opacity": 1,
          "width": "25%",
        }
      }
    >
      <Text
        adjustsFontSizeToFit={true}
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 45,
              "fontWeight": "200",
              "textAlign": "center",
            },
          ]
        }
      >
        3
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#fa9f0d",
          "borderColor": "#303030",
          "borderStyle": "solid",
          "borderWidth": 1,
          "justifyContent": "center",
          "opacity": 1,
          "width": "25%",
        }
      }
    >
      <Text
        adjustsFontSizeToFit={true}
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 45,
              "fontWeight": "200",
              "textAlign": "center",
            },
          ]
        }
      >
        +
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#707070",
          "borderColor": "#303030",
          "borderStyle": "solid",
          "borderWidth": 1,
          "justifyContent": "center",
          "opacity": 1,
          "width": "50%",
        }
      }
    >
      <Text
        adjustsFontSizeToFit={true}
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 45,
              "fontWeight": "200",
              "textAlign": "center",
            },
          ]
        }
      >
        0
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#707070",
          "borderColor": "#303030",
          "borderStyle": "solid",
          "borderWidth": 1,
          "justifyContent": "center",
          "opacity": 1,
          "width": "25%",
        }
      }
    >
      <Text
        adjustsFontSizeToFit={true}
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 45,
              "fontWeight": "200",
              "textAlign": "center",
            },
          ]
        }
      >
        .
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#fa9f0d",
          "borderColor": "#303030",
          "borderStyle": "solid",
          "borderWidth": 1,
          "justifyContent": "center",
          "opacity": 1,
          "width": "25%",
        }
      }
    >
      <Text
        adjustsFontSizeToFit={true}
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 45,
              "fontWeight": "200",
              "textAlign": "center",
            },
          ]
        }
      >
        =
      </Text>
    </View>
  </View>
</View>
`;
